### DISCUSSION TITLE
Reporting errors

### DISCUSSION THREAD

**Discussion by petermr** [petermr/petermr#35] [Announcements]
It's very important to help readers when you report errors. We frequently get:
* "The file didn't work"
* "the program crashed"
etc.

This doesn't help. We need to know:
* your environment ("Windows11", "GoogleColab", etc.)
* what you did ("running installed `docanalysis` on commandline with no arguments).
<give inout command> in code format
* what you were expecting to happen ("expecting a help menu")
* what happened.
<give output in code format> Truncate if too long
* any other info ("the same command worked on <date>")

<more later>

## resources
https://www.chiark.greenend.org.uk/~sgtatham/bugs.html

**Comment by EmanuelFaria** [comment]
REPORT ON: Eliminating possible issues/errors/conflicts related to outdated python versions 

ISSUE:
I have experienced errors when either installing or running [pygetpapers, docanalysis and py4ami](https://github.com/petermr/.) Some of those errors may be unique to me (i.e. not reported by others), some not.[1]

Having been made aware that some of the programs developed by our group specifically require that the latest version of Python (currently 3.10) is installed and properly configured as the default on a my computers:[2] 

1. I installed version Python3.10.4 successfully (no errors reported).
2. I ran `python3 -V` to double-check that the version was indeed installed correctly
3. `python3 -V` consistently returned 3.9 as the installed/active version — even after repeated attempts and diverse methods (updating, re-installing, and force re-installing).

AIMS:
1. To find a solution that will eliminate any possibility that errors I’ve encountered when installing and/or running  may be related to python3 version issues (i.e. defaulting to and running any version of python other than that which is prerequisite for the proper functioning of these programs).[2]
2. Share the solution with others that may similarly be affected. 


ACTION TAKEN
I uninstalled all outdated versions of python@3x using the following brew commands:

`brew uninstall python@3.9` —> python 3.9 was successfully uninstalled
`brew uninstall python@3.8`—> "Error: No such keg: /usr/local/Cellar/python@3.9"
`brew uninstall python@3.7` —> Error: No such keg: /usr/local/Cellar/python@3.8"
(brew did not recognize any command below 3.7)


RESULT
Following this, running `python3 -V` returned the correct/latest installed version.

SUGGESTED ACTION
If deemed appropriate, publish this solution somewhere in the installation instructions for any programs that will not run on python 3.9 or below.


[1]This issue may very well be specific to me, or it could be a bug in python’s installation or upgrade that effects some other mac users that may use these programs.

[2]  had these issues and followed the same steps on both of my Macbooks:
- 2019 Macbook Pro running MacOS v 12.5 Beta (21G5037d)
- 1012 Macbook Air running MacOS v 10.15.7
